strapi/strapi
# Fix plugin config per environment doesn't apply

Signed-off-by: harimkims <harimkims@gmail.com>



### What does it do?

I added logic to find `config/env/<process.env.NODE_ENV>/plugins.js` first when it loads plugin config.
**As-is**: `config/plugins.js` -> default plugin config
**To-be**: `config/env/${process.env.NODE_ENV}/plugins.js` -> `config/plugins.js` -> default plugin config

### Why is it needed?

To support [Environment configurations](https://docs.strapi.io/developer-docs/latest/setup-deployment-guides/configurations/optional/environment.html#environment-configurations)

### How to test it?

1. Create `config/env/development/plugins.js`
2. Copy and taste config below
```js
module.exports = () => ({
  upload: {
    enabled: true,
    config: {
      provider: 'local',
      providerOptions: {
        sizeLimit: 3000000,
      },
    },
  },
});
```
3. Start interactive CLI `yarn strapi console`
4. Try `strapi.config.get('plugin.upload', {})` and see difference
5. Create `config/plugins.js`
6. Repeat Step 2~4 but change `sizeLimit` to 2000000
7. Delete `config/plugins.js`
8. Repeat Step 3~4 (default `sizeLimit` is 1000000 though)

### Related issue(s)/PR(s)

Closes #11774
